frontend/src/js/reducers/Reducer.js

```diff
@@ -35,96 +55,198 @@ var default_sheets_model = {
     sheets: [default_sheet]
 }
 
-const sheetReducer = (sheet = default_sheet, action) => {
-    switch(action.type){
-        case 'ROW_COUNT_CHANGED':
-            sheet.row_count = action.count;
-            break;
-        case 'COL_COUNT_CHANGED':
-            sheet.col_count = action.count;
-            break;
+const moveRow = (cells, action) => {
+    var item = cells.splice(action.dragged_index, 1)[0];
+
+    if (action.position === 'top') {
+        cells.splice(action.dropped_index - 1, 0, item)
+    } else {
+        cells.splice(action.dropped_index, 0, item)
     }
 }
 
+const moveCol = (cells, action) => {
+    cells.forEach((row, i) => {
+        var item = row.splice(action.dragged_index, 1)[0];
+
+        if (action.position === 'left') {
+            row.splice(action.dropped_index - 1, 0, item)
+        } else {
+            row.splice(action.dropped_index, 0, item)
+        }
+    });
+}
+
+const createCellGrid = (sheet) => {
+    var cells = sheet.cells;
+    var row_count = sheet.row_count;
+    var col_count = sheet.col_count;
+    var rows = []
+    for (var i = 0; i < row_count; i++) {
+        let col = []
+        for (var j = 0; j < col_count; j++) {
+            if (i < cells.length && j < cells[i].length) {
+                col.push(cells[i][j]);
+            } else {
+                col.push(second_cell);
+            }
+        }
+        rows.push(col);
+    }
+    //update selected cells
+    var new_arr = [];
+    sheet.selected_cell_indices.forEach((value, index, arr) => {
+        if (!value[0] >= row_count) {
+            new_arr.push(value);
+        }
+        if (!value[1] >= col_count) {
+            new_arr.push(val);
+        }
+    });
+    if (new_arr.length < 1) {
+        new_arr = [
+            [-1, -1]
+        ];
+    }
+    sheet.selected_cell_indices = new_arr;
+    return rows
+}
+
 const varListReducer = (state = test_vars, action) => {
     switch (action.type) {
-        default:
-            return state
+        default: return state
     }
 }
 
 const gmListReducer = (state = test_gms, action) => {
     switch (action.type) {
-        default:
-            return state
+        default: return state
     }
 }
 
 const templateListReducer = (state = test_temps, action) => {
     switch (action.type) {
-        default:
-            return state
+        default: return state
     }
 }
 
 const updateCell = (cell, action) => {
```

Comment:
... this is super weird syntax. Isn't this exactly the same as `function updateCell(cell, action) {`?